File: Maps/CountNicePairs/count.cpp
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int reverseDigit(int n){
        int ans = 0;
        while(n){
            ans *= 10;
            ans += n%10;
            n /= 10;
        }
        return ans;
    }
    int countNicePairs(vector<int>& nums) {
        int n = nums.size();
        for(int i=0;i<n;i++){
            nums[i] -= reverseDigit(nums[i]);
        }
        int count = 0;
        unordered_map<int,int>mp;
        for(int i=0;i<n;i++){
            count = count% 1000000007;
            if(mp.find(nums[i]) != mp.end()){
                count += mp[nums[i]];
            }
            mp[nums[i]]++;
        }
        return count%1000000007;
    }
};

File: Maps/CountNicePairs/count.py
from typing import List

class Solution:
    def reverseDigit(self, n):
        return int(str(n)[::-1])
    def countNicePairs(self, arr: List[int]) -> int:
        n = len(arr)
        for i in range(n):
            arr[i] -= self.reverseDigit(arr[i])
        count = 0
        mp = {}
        for i in range(n):
            count = count%1000000007
            if arr[i] in mp:
                count += mp[arr[i]]
            mp[arr[i]] = mp.get(arr[i],0) + 1
        return count%1000000007